aces/Loris
# [Core] Move PSCID generation config settings from config file to Configuration module

## Brief summary of changes

Moves settings related to generated PSCIDs and ExternalIDs to the Config module instead of the config.xml table.

This should simplify the code and make it easier for users to configure LORIS without the need for a developer.

#### Details

* Adds new patch for new config settings
* Updates SiteIDGenerator to grab config settings from DB instead of config.xml
* Refactors some of the Config module to be more modern/clean
* Adds function `Utility::toDropdownArray` to simplify the creation of arrays like:

```php
array(
'' => '',
'option1' => 'option1',
'option2' => 'option2',
...
);
```

#### Testing instructions (if applicable)

1. Checkout the branch and run the new patch
2. Go to the configuration module and verify that you can change settings related to ID generation under the Study heading
3. Create new candidates using various config settings and ensure that the values make sense given your settings. E.g. IDs created with the `numeric` alphabet should only contain numbers.

#### Links to related tickets (GitHub, Redmine, ...)

* Resolves #5292
